drivers/rtc/rtc_counter_rtc.c

```diff
@@ -0,0 +1,330 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT counter_rtc
+
+#include <zephyr/drivers/rtc.h>
+#include <zephyr/drivers/counter.h>
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/types.h>
+#include <zephyr/logging/log.h>
+#include <time.h>
+#include <fsl_rtc.h>
+#include "rtc_utils.h"
+
+LOG_MODULE_REGISTER(counter_rtc, CONFIG_RTC_LOG_LEVEL);
+
+struct counter_rtc_config {
+	const struct device *counter_dev;
+};
+
+struct counter_rtc_data {
+#ifdef CONFIG_RTC_ALARM
+	const struct device *rtc_dev;
+	rtc_alarm_callback alarm_callback;
+	void *alarm_user_data;
+	uint16_t alarm_mask;
+	bool alarm_pending;
+#endif /* CONFIG_RTC_ALARM */
+};
+
+/* Generic RTC time to ticks conversion using time.h and counter API */
+static uint32_t counter_rtc_time_to_ticks(const struct rtc_time *timeptr)
+{
+	struct tm tm_val = {
+		.tm_sec = timeptr->tm_sec,
+		.tm_min = timeptr->tm_min,
+		.tm_hour = timeptr->tm_hour,
+		.tm_mday = timeptr->tm_mday,
+		.tm_mon = timeptr->tm_mon,
+		.tm_year = timeptr->tm_year,
+		.tm_isdst = -1,
+	};
+
+	time_t seconds = mktime(&tm_val);
+
+	if (seconds == (time_t)-1) {
+		return 0;
+	}
+
+	return (uint32_t)seconds;
+}
```

Comment:
This will only work for counters with a tick rate of 1 hertz, please add a check somewhere to ensure the counter is actually running at 1 hertz, maybe the init function using get_freq()?